### PR TITLE
refactor(frontend): add DI implementation of RaoidcService

### PR DESCRIPTION
**Pull request by sebastien-comeau**
### Description

This pull request introduces the `RaoidcService` in `~/.server/auth` and integrates it into the dependency injection (DI) system.

### Checklist

- [x] I have tested the changes locally
- [x] I have updated the documentation if necessary
- [x] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/.server/auth/raoidc.service.ts [hunk 1]
```diff
@@ -0,0 +1,238 @@
+/**
+ * The RAOIDC Service module is responsible for all RAOIDC interactions in the
+ * application. It exports a single getRaoidcService() factory function that
+ * should be used to get an instance of the service.
```
Comment (gregory-j-baker) [inline frontend/app/.server/auth/raoidc.service.ts:4]:
```suggestion
 * The RAOIDC Service module is responsible for all RAOIDC interactions in the
 * application.
```

File: frontend/app/.server/auth/raoidc.service.ts [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+ *
+ * If an HTTP proxy is configured (via the HTTP_PROXY_URL environment
+ * variable), the service will make all RAOIDC calls with a custom fetch()
+ * function that uses that HTTP proxy. This is most commonly used during
+ * development, when the nonprod RAOIDC instance is not accessible outside of
+ * the ESDC network.
+ *
+ * The service module memoizes the service instance once it is successfully
+ * initialized. This ensures that only a single instance of the service is
+ * created for the application.
```
Comment (gregory-j-baker) [inline frontend/app/.server/auth/raoidc.service.ts:21]:
```suggestion
 *
 * The service module memoizes the service instance once it is successfully
 * initialized. This ensures that only a single instance of the service is
 * created for the application.
```

File: frontend/app/.server/auth/raoidc.service.ts [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    }
+
+    this.log.debug('Authentication check passed');
+  }
+
+  private moizedFetchServerMetadata = moize(fetchServerMetadata, {
+    onCacheAdd: () => {
+      this.log.info('Creating new fetchServerMetadata memo');
+    },
+  });
```
Comment (gregory-j-baker) [inline frontend/app/.server/auth/raoidc.service.ts:237]: This should have a TTL because the contents of the discovery document could change.

Comment (sebastien-comeau) [reply frontend/app/.server/auth/raoidc.service.ts:237]: I will add a TODO.

**Review by gregory-j-baker — APPROVED** [review]

**Pull request merged**